src/System.Management.Automation/engine/regex.cs

```diff
@@ -278,6 +278,70 @@ public static bool ContainsWildcardCharacters(string pattern)
             return result;
         }
 
+        /// FIXME
+        /// For invalid wildcard pattern, it currently only check if there is
+        /// unclosed bracket. Need to find a better solution and rewrite this.
+        ///
+        /// <summary>
+        /// Checks to see if the given string might contains wildcard pattern.
+        /// </summary>
+        /// <param name="pattern">
+        /// String which needs to be checked
+        /// </param>
+        /// <returns>
+        /// true if the string does not contain invalid wildcard pattern,
+        /// false otherwise.
+        /// </returns>
+        /// <remarks>
+        /// Currently string contains { '*', '?' } or both { '[', ']' } is
+        /// considered to be properbly a wildcard pattern and
+        /// '`' is the escape character.
+        /// </remarks>
+        internal static bool ContainsValidWildcardPattern(string pattern)
```

Comment:
A couple of extra cases this method should reject:

- `"[a-mt]ook"`
- `"[ab-z]ook"`
- `"[a?b]"`
- `"[a*]x"`
- `"goose]"`

Reply:
Worth putting these into test cases

Reply:
I'm confused now. All patterns suggested in the list are valid, would you please explain why should `ContainsValidWildcardPattern` reject them?

Reply:
`"goose]"` is a not a pattern, even though it's valid (not throw when calling `WildCardPatter.IsMatch`)